### PR TITLE
Fix x86 clrgc loading

### PR DESCRIPTION
**Pull request by cshung**
The loading of the `clrgc.dll` is currently broken because the `GetProcAddress` call failed here.

https://github.com/dotnet/runtime/blob/daeb683623f4539711b851cbc49b5b58a79ca0bf/src/coreclr/vm/gcheaputilities.cpp#L212

The call failed because the function name is mangled, and using the `__cdecl` calling convention fixed it.

### CODE REVIEW

**Review by PeterSolMS — APPROVED** [review]
Looks good to me!

File: docs/design/features/standalone-gc-loading.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ struct VersionInfo {
   const char* Name;
 };
 
-extern "C" void GC_VersionInfo(
+extern "C" void __cdecl GC_VersionInfo(
```
Comment (jkotas) [inline docs/design/features/standalone-gc-loading.md:113]:
```suggestion
extern "C" void LOCALGC_CALLCONV GC_VersionInfo(
```

File: docs/design/features/standalone-gc-loading.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ Once the EE has verified that the version of the candidate GC is valid, it then
 GC. It does so by loading (via `GetProcAddress`) and executing a function with this signature:
 
 ```c++
-extern "C" HRESULT GC_Initialize(
+extern "C" HRESULT __cdecl GC_Initialize(
```
Comment (jkotas) [inline docs/design/features/standalone-gc-loading.md:145]:
```suggestion
extern "C" HRESULT LOCALGC_CALLCONV GC_Initialize(
```

File: src/coreclr/gc/sample/GCSample.cpp [hunk 3]
```diff
@@ -47,6 +47,12 @@
 
 #include "gcdesc.h"
 
+#ifdef TARGET_X86
+#define CALLCONV __cdecl
```
Comment (jkotas) [inline src/coreclr/gc/sample/GCSample.cpp:51]:
```suggestion
#define LOCALGC_CALLCONV __cdecl
```

File: src/coreclr/gc/sample/GCSample.cpp [hunk 4]
```diff
@@ -47,6 +47,12 @@
...
+#else
+#define CALLCONV
```
Comment (jkotas) [inline src/coreclr/gc/sample/GCSample.cpp:53]:
```suggestion
#define LOCALGC_CALLCONV
```

File: src/coreclr/gc/sample/GCSample.cpp [hunk 5]
```diff
@@ -106,9 +112,9 @@ void WriteBarrier(Object ** dst, Object * ref)
     ErectWriteBarrier(dst, ref);
 }
 
-extern "C" HRESULT GC_Initialize(IGCToCLR* clrToGC, IGCHeap** gcHeap, IGCHandleManager** gcHandleManager, GcDacVars* gcDacVars);
+extern "C" HRESULT CALLCONV GC_Initialize(IGCToCLR* clrToGC, IGCHeap** gcHeap, IGCHandleManager** gcHandleManager, GcDacVars* gcDacVars);
 
-int __cdecl main(int argc, char* argv[])
+int CALLCONV main(int argc, char* argv[])
```
Comment (jkotas) [inline src/coreclr/gc/sample/GCSample.cpp:117]:
```suggestion
int __cdecl main(int argc, char* argv[])
```
This should stay as-is. It is not unrelated to LOCALGC_CALLCONV

File: src/coreclr/gc/sample/GCSample.cpp [hunk 6]
```diff
@@ -106,9 +112,9 @@ void WriteBarrier(Object ** dst, Object * ref)
     ErectWriteBarrier(dst, ref);
 }
 
-extern "C" HRESULT GC_Initialize(IGCToCLR* clrToGC, IGCHeap** gcHeap, IGCHandleManager** gcHandleManager, GcDacVars* gcDacVars);
+extern "C" HRESULT CALLCONV GC_Initialize(IGCToCLR* clrToGC, IGCHeap** gcHeap, IGCHandleManager** gcHandleManager, GcDacVars* gcDacVars);
```
Comment (jkotas) [inline src/coreclr/gc/sample/GCSample.cpp:115]:
```suggestion
extern "C" HRESULT LOCALGC_CALLCONV GC_Initialize(IGCToCLR* clrToGC, IGCHeap** gcHeap, IGCHandleManager** gcHandleManager, GcDacVars* gcDacVars);
```

**Pull request merged**